docs/quickstart.md

```diff
@@ -0,0 +1,185 @@
+in a lakeFS repository using `lakefs-spec`.
+
+??? info "Prerequisites"
+
+    To follow along with this guide, you will need a few prerequisites ready on your machine:
+
+    - `lakefs-spec` supports Windows, macOS, or Linux
+    - [Docker](https://www.docker.com/), with Docker Compose (although Podman should work as well, but is untested)
+    - [Python 3.9](https://python.org) or later
+    - optionally, [`lakectl`](https://docs.lakefs.io/reference/cli.html), the lakeFS command line tool
+
+    Please take a moment to make sure you have these tools available before proceeding with the next steps.
+
+## Installing `lakefs-spec`
+
+??? tip "A note on virtual environments"
+
+    We generally recommend installing the library in a virtual environment to ensure proper isolation, especially when following this quickstart guide.
+
+    If you are using Poetry, virtual environments can automatically be created by the tool.
+
+    If you prefer the `venv` functionality built into Python, see the [official docs](https://docs.python.org/3/library/venv.html) (tl;dr: `python -m venv venv; source venv/bin/activate`).
+
+`lakefs-spec` can be used on any platform and requires at least Python 3.9.
+
+To install the package directly from PyPI, run:
+
+=== "pip"
+
+    ```
+    pip install lakefs-spec
+    ```
+
+=== "poetry"
+
+    ```
+    poetry add lakefs-spec
+    ```
+
+Or, if you want to try the latest pre-release version directly from GitHub:
+
+=== "pip"
+
+    ```
+    pip install git+https://github.com/appliedAI-Initiative/lakefs-spec.git
+    ```
+
+=== "poetry"
+
+    ```
+    poetry add git+https://github.com/appliedAI-Initiative/lakefs-spec.git
+    ```
+
+## First Steps
+
+### Spinning up a local lakeFS instance
+
+!!! warning
+    This setup is not recommended for production uses, since it does not store the data persistently.
+
+    Please check out the [lakeFS docs](https://docs.lakefs.io/howto/deploy/) for production-ready deployment options.
+
+If you don't already have access to a lakeFS server, you can quickly start a local instance using Docker Compose with a [configuration file](https://github.com/appliedAI-Initiative/lakefs-spec/blob/main/hack/docker-compose.yml) provided in the `lakefs-spec` repository:
+
+```shell
+$ curl https://raw.githubusercontent.com/appliedAI-Initiative/lakefs-spec/main/hack/docker-compose.yml | docker-compose -f - up
+```
+
+If you do not have `curl` installed on your machine or would like to examine and/or customize the container configuration, you can also create a `docker-compose.yml` file locally and use it with `docker-compose up`:
+
+```yaml title="docker-compose.yml"
+--8<-- "https://raw.githubusercontent.com/appliedAI-Initiative/lakefs-spec/main/hack/docker-compose.yml:3:"
+```
+
+In order to allow `lakefs-spec` to automatically discover credentials to access this lakeFS instance, create a `.lakectl.yaml` in your home directory containing the following:
```

Comment:
This `.lakectl.yaml` is also created by using lakectl during the setup via `lakectl config`

Reply:
Added in f6ae065f62e5d194074c9936550688681a275c0e